src/app/coin/[id]/page.tsx

```diff
@@ -0,0 +1,308 @@
+"use client";
+import { useEffect } from "react";
+import { useDispatch } from "react-redux";
+import Image from "next/image";
+import { Progress } from "../../components/ui/progress";
+import { AppDispatch, useAppSelector } from "../../../redux/store";
+import { getCoinPageData } from "../../../redux/features/coinPageSlice";
+import { ChevronUpIcon } from "../../icons/ChevronUpIcon";
+import { ChevronDownIcon } from "../../icons/ChevronDownIcon";
+import formatNumber from "../../../app/utils/formatNumber";
+import convertToHyperlink from "../../../app/utils/convertToHyperlink";
+import HomepageLinks from "../../../app/components/HomePageLinks";
+import BlockchainLinks from "../../../app/components/BlockChainLinks";
+import { PlusIcon } from "../../icons/PlusIcon";
+
+export default function CoinPage({ params }: { params: { id: string } }) {
```

Comment:
Let's define type or interface above this and use that type or interface here.  Instead of writing `{ params: { id: string } }` directly.